### PR TITLE
Restrict admin user from provider level routes.

### PR DESCRIPTION
**Pull request by jessethach**
#### Description

While non-admin users are currently prevented from accessing all admin views in any capacity, the reverse is not actually true currently. While OEC admins are navigated to their default overview page by default (and don't have access to the roster navbar directly), they're also able to access all provider roster views as well, if they know the direct URL.

It should be noted that both the corresponding non-admin views and the non-admin APIs should be appropriately gated by role.

#### Work

- Add logic to AdminRedirect file
- Add admin flag object to differentiate route types
- Sort routes file

## GitHub Issue
#1316

## Associated PRs
None

#### Validate for unauthorized routes (Do not log in)

- [ ] Confirm that user can navigate to following routes: `/template`, `/data-requirements`, `/privacy`, `/help`, `/support`, `/funding-space-types` 
- [ ] Confirm that user can navigate to `/login` page
- [ ] Sign in to either type of user role. Then logout. Confirm that user is navigated to landing page where it asked user to sign in.

#### Validate for Admin level routes

- [ ] Confirm that user can navigate to all admin level routes (check `client/src/routes.ts` for reference)
- [ ] Confirm that user will be redirected to `/overview` route if attempting to navigate to non admin routes (check `client/src/routes.ts` for reference)

#### Validate for Provider level routes

- [ ] Confirm that user can navigate to all provider level routes (check `client/src/routes.ts` for reference)
- [ ] Confirm that user will be redirected to `/home` route if attempting to navigate to admin only routes (check `client/src/routes.ts` for reference)

## Tech Specs
N/A

## Automated Testing
- [ ] All unit tests are passing.
- [ ] Unit tests have been added to cover this changeset.

## Additional Context
N/A

### CODE REVIEW

**Review by lukenewman — APPROVED** [review]

**Review by gordonfarrell — APPROVED** [review]
Great stuff! Looks good 🏄🏼‍♂️

**Comment by gordonfarrell** [comment]
Jumping on validation for this

*****************************************************************
## Validation Plan

#### Validate for unauthorized routes (Do not log in)

- [ ] Confirm that user can navigate to following routes: `/template`, `/data-requirements`, `/privacy`, `/help`, `/support`, `/funding-space-types` 
- [ ] Confirm that user can navigate to `/login` page
- [ ] Sign in to either type of user role. Then logout. Confirm that user is navigated to landing page where it asked user to sign in.

#### Validate for Admin level routes

- [ ] Confirm that user can navigate to all admin level routes (check `client/src/routes.ts` for reference)
- [ ] Confirm that user will be redirected to `/overview` route if attempting to navigate to non admin routes (check `client/src/routes.ts` for reference)

#### Validate for Provider level routes

- [ ] Confirm that user can navigate to all provider level routes (check `client/src/routes.ts` for reference)
- [ ] Confirm that user will be redirected to `/home` route if attempting to navigate to admin only routes (check `client/src/routes.ts` for reference)

**Comment by gordonfarrell** [comment]
Validated and good to go! 🚀

**Comment by jessethach** [comment]
@nickbristow ready for merge!

File: client/src/routes.ts [hunk 1]
```diff
@@ -26,166 +26,233 @@ import Organizations from './containers/Organizations/Organizations';
...
 import { RevisionRequest } from './containers/Home/RevisionRequest';
 
+export type AdminFlag =
+  | {
+      adminOnly: true;
+      restrictAdmin?: false;
+    }
+  | {
+      adminOnly?: false;
+      restrictAdmin: true;
```
Comment (SMakaiTakori) [inline client/src/routes.ts:36]: Looks great! Did you want to include the `?` like you did for line 32?

**Review by SMakaiTakori — APPROVED** [review]
Looks good!

File: client/src/routes.ts (hunk 1 above)
Comment (SMakaiTakori) [inline client/src/routes.ts:36]: Looks good! Did you want to include the `?` like on line 32?

**Pull request merged**